### ISSUE TITLE
Add the "Pot without detent" selection that S1 and S2 have to LS and RS sliders.

### ISSUE THREAD

**Issue by Miami-Mike** [opentx/opentx#8643]
I replaced the LS pot on my Taranis X9D+ with a pushbutton and can't calibrate it properly because there's no way to center it. Now, with recent Companion versions, I can't even work around that by editing the calibration settings.

But unless I'm mistaken, if the LS and RS pots had "Pot without detent" settings like S1 and S2 have then I could get the full -100% to +100% range without centering during calibration. So whether you restore the ability to edit calibration settings or not, I think it would be good for LS and RS to have "Pot without detent" options like S1 and S2 have.

**Comment by 3djc** [comment]
Makes sense, I just hope there isn’t too much hard coded to save eeprom, will have a look

**Comment by rotorman** [comment]
@Miami-Mike There might be a hardware solution to your issue. You could add two resistors to the analog channel with the switch in a voltage divider circuit fashion. The value of the resistors is not very relevant, it should not be too low for high current draw, but also not too high for better noise filtering, e.g. 2x 4K7 would be a good, but 2x 1K or 2x 10K would also work.

One resistor from signal line to AGND, other from signal line to AVDD.

Then your switch pulling the signal line either to AGND or AVDD, does not matter. If you let the switch open, you have center position. If the switch closes, you have one extreme.

You could even add a 3-pos switch, one switch end would pull to AGND, other to AVDD.

**Comment by Miami-Mike** [comment]
It's a pushbutton with two states. You're either pushing it or you're not.

![button](https://user-images.githubusercontent.com/22988476/127739319-fbcdbc03-54fb-4569-b913-807fac96f544.jpg)

You might be able to see in the picture that I also replaced S1 and S2 with 3-position switches. They're set up with resistor voltage dividers and after calibration the three positions return values of -100%, 0%, and +100%. S1 and S2 are set as "Pots with detent" on the Hardware page.

I discovered by experimentation with another OpenTX radio that, apparently, when S1 or S2 are set to "Pot without detent" then during the first step of calibration the center position is ignored. When S1 and S2 are set like that, it appears that only the end limits are considered and they return values linearly spaced from -100% to +100% with center having no significance.

That's what I need for LS. My experiments show that if S1 and/or S2 were replaced by pushbuttons and set for "Pot without detent" then they could be made to return values of -100% and +100%, but LS and RS don't have that.

**Comment by rotorman** [comment]
As you are customizing hw, why not just fork the source code and do the change in code as you like. You can change the LS and RS to have the functionality you like.
Building OpenTX from source is also not very daunting task, have a look at https://github.com/opentx/opentx/wiki/Compiling-OpenTx

**Comment by 3djc** [comment]
What type of x9d @Miami-Mike ?

**Comment by Miami-Mike** [comment]
It's a FrSky Taranis X9D+, but I think this principle really should apply to OpenTX in general. When it's possible for the center position of a pot to lack significance, whether due to a modification or its assigned function, there should be an option to set it to "Pot without detent" so that its calibration is based solely upon its endpoints.

**Comment by 3djc** [comment]
Could you check this works ok ?

[firmware.zip](https://github.com/opentx/opentx/files/6912324/firmware.zip)

It should give you a new sliderwithout detent choice, and should work (but I hve no x9d of any kind to test with)

**Comment by Miami-Mike** [comment]
I successfully installed the firmware file...

![screen-1970-01-01-000933](https://user-images.githubusercontent.com/22988476/127773724-31604e38-18de-49a1-9611-d70e7b21a593.png)

but I still only get "Slider" and "None" as choices for LS and RS...

![screen-1970-01-01-000138](https://user-images.githubusercontent.com/22988476/127773755-9623f050-335c-4488-ac4b-e467eda7e601.png)

![screen-1970-01-01-000142](https://user-images.githubusercontent.com/22988476/127773758-b60aa801-3953-4d0c-bfe7-365f3ef19ab6.png)

and calibration still does the same thing. LS and RS behave the way S1 and S2 do when they're set to "Pot with detent", not the way they behave when set to "Pot without detent".

The firmware you uploaded here appears to work the same as the 2.3.14 firmware I already have, aside from the different built options.

**Comment by 3djc** [comment]
I probably uploaded the wrong file then, let me check

**Comment by 3djc** [comment]
Here you go

![image](https://user-images.githubusercontent.com/5167938/127774973-36500d59-30c8-4a20-a829-9cbefad7b34a.png)

[firmware.zip](https://github.com/opentx/opentx/files/6912832/firmware.zip)

**Comment by Miami-Mike** [comment]
With your second version the "Slider w. det" option shows on my radio screen just like in your photo but there's no change in behavior.

If "Slider w. det" is meant to be the same thing as "Pot with detent" then that's not what I'm looking for, and in fact is no different from the way the slider already behaves. What I'm looking for is the equivalent of "Pot without detent", which on the Taranis X9D+ radio screen shows as simply "Pot".

![LS-RS](https://user-images.githubusercontent.com/22988476/127775920-66948509-4472-447b-9ba9-d43f277f63c4.png)

I really appreciate all of the effort you're putting into this!

**Comment by 3djc** [comment]
You obviously won't get Pot for a slider,, but "slider"entry should behave like pot does,

**Comment by 3djc** [comment]
No, can't be done wihtout chaning EEprom, we have only 1 bit for sliders; should have checked that first. Need to wait 2.4

**Comment by Miami-Mike** [comment]
"Slider" currently behaves like "Pot with detent". If you change the label "Slider" to mean behavior like "Pot" (which is "Pot without detent" in Companion) then you'll change how some model setups behave. In some setups the "with detent" setting may be important because that setting ensures that the center position, as indicated by the detent, returns a value close to 0%. In other setups the center position is not important, such as when the potentiometer is used to control volume.

So it'll be a matter of choosing naming conventions and default settings that won't cause new problems. The name you chose, "Slider w. det." would work as a replacement name for what the current "Slider" setting does, but you wouldn't want "Slider" to be redefined as the setting where center is not significant (like the "Pot" setting).

Personally, I don't see the point in differentiating between the potentiometers on top that are controlled with knobs (S1 & S2) and the side potentiometers that are controlled with levers (LS & RS). They're all potentiometers, so why not provide the same setting options and associated labels for LS and RS that currently exist for S1 and S2?    

Anyway, thanks again for your effort!

**Comment by 3djc** [comment]
They are diffrents. Pots are CW, sliders toward top/center, but thats not the point, pots have 2 bits storage per pot (allowing 3 choi ces), while sliders only have one, and that cannot be changed without EEprom changes as I said, so 2.4

**Comment by Miami-Mike** [comment]
I understand.
